### ISSUE TITLE
TypeError: Cannot read property '*' of undefined

### ISSUE THREAD

**Issue by majidhassan** [parse-community/parse-server#7456]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description

I just upgrade to parse-server 4.5.0 and I'm getting the error below (in the logs section).

I'm running this using docker-compose, the parse container keeps restarting a couple of times with this error, then seems to be working fine. I checked the [changelog](https://github.com/parse-community/parse-server/blob/master/CHANGELOG.md#402) and specifically [#6465](https://github.com/parse-community/parse-server/issues/6465) but that doesn't seem to be it, since I'm running parse-server 4.5.0 and mongodb 4.4.6.

### Steps to reproduce

Not sure how to reproduce this, because I'm pretty sure not everyone using parse-server 4.5.0 has this issue.

### Actual Outcome

Server keeps restarting.

### Expected Outcome

Server starts up successfully first time I start it.

### Failing Test Case / Pull Request

- [ ] 🤩 I submitted a PR with a fix and a test case.
- [ ] 🧐 I submitted a PR with a failing test case.

###  Environment

Server
- Parse Server version: `4.5.0`
- Operating system: `Ubuntu 20.04 LTS`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `Digital Ocean`

Database
- System (MongoDB or Postgres): `MongoDB`
- Database version: `4.4.6`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `MongoDB Atlas`

Client (irrelevant as server doesn't start up)
- SDK (iOS, Android, JavaScript, PHP, Unity, etc): `FILL_THIS_OUT`
- SDK version: `FILL_THIS_OUT`

### Logs

```
warn: Unable to ensure uniqueness for usernames:  Cannot read property '*' of undefined {"stack":"TypeError: Cannot read property '*' of undefined\n    at SchemaData.get (/app/node_modules/parse-server/lib/Controllers/SchemaController.js:671:84)\n    at SchemaController.enforceClassExists (/app/node_modules/parse-server/lib/Controllers/SchemaController.js:1013:24)\n    at /app/node_modules/parse-server/lib/Controllers/DatabaseController.js:1432:70\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)"}
warn: Unable to create case insensitive username index:  Cannot read property '*' of undefined {"stack":"TypeError: Cannot read property '*' of undefined\n    at SchemaData.get (/app/node_modules/parse-server/lib/Controllers/SchemaController.js:671:84)\n    at SchemaController.enforceClassExists (/app/node_modules/parse-server/lib/Controllers/SchemaController.js:1013:24)\n    at /app/node_modules/parse-server/lib/Controllers/DatabaseController.js:1432:70\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)"}
warn: Unable to ensure uniqueness for user email addresses:  Cannot read property '*' of undefined {"stack":"TypeError: Cannot read property '*' of undefined\n    at SchemaData.get (/app/node_modules/parse-server/lib/Controllers/SchemaController.js:671:84)\n    at SchemaController.enforceClassExists (/app/node_modules/parse-server/lib/Controllers/SchemaController.js:1013:24)\n    at /app/node_modules/parse-server/lib/Controllers/DatabaseController.js:1432:70\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)"}
warn: Unable to create case insensitive email index:  Cannot read property '*' of undefined {"stack":"TypeError: Cannot read property '*' of undefined\n    at SchemaData.get (/app/node_modules/parse-server/lib/Controllers/SchemaController.js:671:84)\n    at SchemaController.enforceClassExists (/app/node_modules/parse-server/lib/Controllers/SchemaController.js:1013:24)\n    at /app/node_modules/parse-server/lib/Controllers/DatabaseController.js:1432:70\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)"}
TypeError: Cannot read property '*' of undefined
    at SchemaData.get (/app/node_modules/parse-server/lib/Controllers/SchemaController.js:671:84)
    at SchemaController.enforceClassExists (/app/node_modules/parse-server/lib/Controllers/SchemaController.js:1013:24)
    at /app/node_modules/parse-server/lib/Controllers/DatabaseController.js:1432:70
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

**Comment by mtrezza** [comment]
Thanks for reporting.

Do I understand correctly that this happened only once, you never observed this anymore, and you are not able to reproduce it, because you don@have the issue-causing image anymore?

**Comment by majidhassan** [comment]
@mtrezza hey, thanks for replying.

The issue happened several times. I could give you a bit more context, so I have a snapshot of my server on Digital Ocean, and I spawned a droplet from that snapshot. Whenever the droplet spins up (and the docker container is created) I get the error mentioned above for about three to four times (which causes the parse container to restart) and then all is good.

This happened every time I spawned a droplet, but never happened locally even though I was using the same Docker image.

**Comment by mtrezza** [comment]
Could you to look at your DB and see whether there are any changes regarding your indices in the User collection after each restart? You could also look into your DB logs or add a log entry for the Parse Server DB adapter to see what the detailed error message is.

**Comment by majidhassan** [comment]
@mtrezza thanks for your reply. I just tried to reproduce, and I stopped getting this error. However, I checked the indexes, and it doesn't seem like new indexes are added with each restart.

**Comment by mtrezza** [comment]
Thanks for the update. I understand that you could not reproduce the issue anymore. I'm closing this as it seems to be resolved. Feel free to comment if you have any questions and we can re-open this issue.

**Comment by malcolm-dane** [comment]
I'm also having this issue! I can't figure out how to fix it. Can't run Cloud Code Queries, dropped version down to 4.10.0.

**Comment by mtrezza** [comment]
@malcolm-dane If you can reliably reproduce the issue, then we can look into it.

**Comment by malcolm-dane** [comment]
> @malcolm-dane If you can reliably reproduce the issue, then we can look into it.

I'm not too sure what exactly is causing it, but it's related to something in the middleware. I've received the schema error, and a related data basecontroller error. I can't do cloud functions that search database items, I've gotten the same errors as above, and another error related to the database controller. 

I am planning on building the server from source, deploying it and seeing if the error persists. It appears that useMasterKey is ignored, along with requireUser set to false being off. 

Any ideas what can be causing this? If I remember correctly version 3 didn't have this issue.

**Comment by mtrezza** [comment]
I am not sure how to support at this point as I don't fully understand the issue. This may be related to your other issue https://github.com/parse-community/parse-server/issues/7728, both of which lack a specific example to reproduce the issue. Maybe if you debug Parse Sever, set some breakpoints and look into what is causing these issues, you get some hints.
